colossus-docs/src/main/scala/GenericClient.scala

```diff
@@ -0,0 +1,50 @@
+import akka.actor.ActorSystem
+import colossus.IOSystem
+import colossus.protocols.http.HttpMethod.Get
+import colossus.protocols.http.UrlParsing.{Root, on}
+import colossus.protocols.http.server.{HttpServer, Initializer, RequestHandler}
+import colossus.protocols.http.{Http, HttpClient, HttpRequest}
+import colossus.service.GenRequestHandler.PartialHandler
+import colossus.service.{Async, CallbackAsync, FutureAsync}
+
+import scala.language.higherKinds
+
+object GenericClient extends App {
+
+  // #example
+  implicit val actorSystem = ActorSystem()
+  implicit val ioSystem = IOSystem()
+
+  def doTheThing[A[_]](client: HttpClient[A])(implicit async: Async[A]): A[String] = {
+    async.map(client.send(HttpRequest.get("/"))) { response =>
+      response.body.bytes.utf8String
+    }
+  }
+
+  // using future client
+  implicit val futureAsync = new FutureAsync()
```

Comment:
I saw the AsyncBuilder, but I couldn't see how to use it and there was no obvious implicits I should be importing.